src/gui/widgets/typed_debugger.cc

```diff
@@ -0,0 +1,967 @@
+    std::istringstream file(std::string(fileContents), std::ios_base::in);
+    std::string line;
+    while (std::getline(file, line)) {
+        // For data types.
+        std::string name;
+        std::vector<FieldOrArgumentData> fields;
+
+        // For functions.
+        uint32_t address;
+        DebuggerFunction func;
+
+        std::stringstream ss(line);
+        std::string s;
+        int linePartsCounter = 0;
+        while (std::getline(ss, s, ';')) {
+            if (linePartsCounter == 0) {
+                switch (importType) {
+                    case ImportType::DataTypes:
+                        // @todo: implement robust typedef handling.
+                        if (s.front() == '_') {
+                            s.erase(0, 1);
+                        }
+                        name = s;
+                        ++linePartsCounter;
+                        continue;
+                        break;
+                    case ImportType::Functions:
+                        address = std::stoul(s, nullptr, 16);
+                        m_functionAddresses.push_back(address);
+                        ++linePartsCounter;
+                        continue;
+                        break;
+                    default:
+                        break;
+                }
+            }
+            if (importType == ImportType::Functions && linePartsCounter == 1) {
+                func.name = s;
+                ++linePartsCounter;
+                continue;
+            }
+
+            FieldOrArgumentData data;
+            const std::regex data_regex(R"(([\w\s\*\[\]]+),(\w+),(\d+))");
+            std::smatch matches;
+            if (std::regex_match(s, matches, data_regex)) {
+                data.type = matches[1].str();
+                data.name = matches[2].str();
+                data.size = std::stoul(matches[3].str());
+            }
+
+            switch (importType) {
+                case ImportType::DataTypes:
+                    fields.push_back(data);
+                    break;
+                case ImportType::Functions:
+                    func.arguments.push_back(data);
+
+                    ++linePartsCounter;
+                    // Address + name + 4 arguments = 6.
+                    if (linePartsCounter == 6) {
+                        break;
+                    }
+                    break;
+                default:
+                    break;
+            }
+        }
+        switch (importType) {
+            case ImportType::DataTypes:
+                m_structs[name] = fields;
+                m_typeNames.push_back(name);
+                break;
+            case ImportType::Functions:
+                m_functions[address] = func;
+                break;
+            default:
+                break;
+        }
+    }
+
+    if (importType == ImportType::DataTypes) {
+        std::sort(m_typeNames.begin(), m_typeNames.end(), [](std::string left, std::string right) {
```

Comment:
This function signature should be `const std::string & left, const std::string & right`.